addons/api/tests/unit/serializers/host-test.js

```diff
@@ -4,19 +4,48 @@ import { setupTest } from 'ember-qunit';
 module('Unit | Serializer | host', function (hooks) {
   setupTest(hooks);
 
-  test('it exists', function (assert) {
-    let store = this.owner.lookup('service:store');
-    let serializer = store.serializerFor('host');
-
-    assert.ok(serializer);
+  test('it serializes host with type static as expected', function (assert) {
+    assert.expect(1);
+    const store = this.owner.lookup('service:store');
+    const record = store.createRecord('host', {
+      name: 'Host 1',
+      compositeType: 'static',
+      description: 'Description',
+      host_catalog_id: '123',
+      version: 1,
+      address: '188e:68a9:b342:c05c:2595:2f46:499c:759f',
+    });
+    assert.deepEqual(record.serialize(), {
+      name: 'Host 1',
+      description: 'Description',
+      type: 'static',
```

Comment:
could you please explain what is happening here with the compostiteType/type change here?

Reply:
Absolutley. I am not 100% sure what you mean by I think I got it.

- This is a serialization test, understanding serialization as a translation from `ember model` (UI client) to `API`.
- When I create the record (so when I use `compositeType`) I am createing a record on `ember model` (UI client). We have `compositeType` available here.
- When I am asserting, I am taking the previously created `ember model`, serialize it, and then comparing to the expectation.
- As explained in the first step, I would expect `type` here, because the object I am expecting is the one the API will receive. The API does not know what `compositeType` is, just the UI.

Happy to jump on zoom call if you need more clarification about this :)

I use a rule of thumb that helps: (thanks @randallmorey 
- Serialization: Ember data -> API.
- Normalization: API -> Ember data.

Reply:
Yup.  The serialised output is what we send to the API.  The API expects `type`.

Reply:
got it! that makes sense. ty for the explanation!

Reply:
Np @DhariniJeeva took me a while until Randy told me the rule of thumb, my mind was always a mess with creation vs expectation :sweat_smile: